File: app/src/main/java/com/snaptiongame/app/data/models/UserStats.java
package com.snaptiongame.app.data.models;

import com.google.gson.annotations.SerializedName;

/**
 * @author Quang Ngo.
 */

public class UserStats {
    @SerializedName(GAMES_CREATED)
    public int gamesCreated;
    @SerializedName(HIGHEST_GAME_UPVOTE)
    public int highestGameUpvote;
    @SerializedName(CAPTIONS_CREATED)
    public int captionsCreated;
    @SerializedName(CAPTION_UPVOTES)
    public int captionUpvotes;
    @SerializedName(TOP_CAPTION_COUNT)
    public int topCaptionCount;

    public static final String GAMES_CREATED = "gamesCreated";
    public static final String HIGHEST_GAME_UPVOTE = "highestGameUpvote";
    public static final String CAPTIONS_CREATED = "captionsCreated";
    public static final String CAPTION_UPVOTES = "captionUpvotes";
    public static final String TOP_CAPTION_COUNT = "topCaptionCount";


}

File: app/src/main/java/com/snaptiongame/app/data/models/UserStats.kt
package com.snaptiongame.app.data.models

import com.google.gson.annotations.SerializedName

/**
 * @author Quang Ngo
 */

class UserStats {
    @SerializedName(RANK)
    var rank: String? = null
    @SerializedName(GAMES_CREATED)
    var gamesCreated: Int = 0
    @SerializedName(HIGHEST_GAME_UPVOTE)
    var highestGameUpvote: Int = 0
    @SerializedName(CAPTIONS_CREATED)
    var captionsCreated: Int = 0
    @SerializedName(CAPTION_UPVOTES)
    var captionUpvotes: Int = 0
    @SerializedName(TOP_CAPTION_COUNT)
    var topCaptionCount: Int = 0
    @SerializedName(EXP)
    var exp: Int = 0

    companion object {
        const val RANK = "rank"
        const val GAMES_CREATED = "gamesCreated"
        const val HIGHEST_GAME_UPVOTE = "highestGameUpvote"
        const val CAPTIONS_CREATED = "captionsCreated"
        const val CAPTION_UPVOTES = "captionUpvotes"
        const val TOP_CAPTION_COUNT = "topCaptionCount"
        const val EXP = "exp"
    }
}
